### PR TITLE
005 view comp by band

### PR DESCRIPTION
**Pull request by orlagh-mccloskey**
Merging with main so will be changes due to resolving conflicts

### CODE REVIEW

File: src/main/java/org/kainos/groupA/services/BandCompService.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.kainos.groupA.exception.BandIDDoesNotExistException;
+import org.kainos.groupA.models.Competency;
+import org.kainos.groupA.utils.DatabaseConnector;
+
+import javax.ws.rs.Path;
+import java.sql.SQLException;
+import java.util.List;
+
+@Path("/api")
+@Api("User API")
```
Comment (shaunganley) [inline src/main/java/org/kainos/groupA/services/BandCompService.java:14]: these annotations aren't needed in a service class

File: src/main/java/org/kainos/groupA/models/Competency.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    public void setComp_description(String comp_description) {
+        this.comp_description = comp_description;
+    }
+
+    private String comp_name;
+    private String comp_description;
+
+    @JsonCreator
+    public Competency(
```
Comment (shaunganley) [inline src/main/java/org/kainos/groupA/models/Competency.java:27]: you'll probably want to return the band name as well if you want to display it on the UI

File: src/test/java/org/kainos/groupA/service/BandCompServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    @Test
+    void getCompByBandID_shouldThrowBandIDDoesNotExist_whenIDisGreaterThanBandTableCount() throws SQLException, BandIDDoesNotExistException {
+        int id = 200;
+        Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+        Mockito.when(bandCompDao.getCompByBandID(conn,id)).thenThrow(BandIDDoesNotExistException.class);
+        assertThrows(BandIDDoesNotExistException.class, () -> bandCompService.getCompByBandID(id));
+    }
+
+    @Test
+    void getCompByBandID_shouldThrowBandIDDoesNotExist_whenIDisLessThan1() throws SQLException, BandIDDoesNotExistException {
```
Comment (shaunganley) [inline src/test/java/org/kainos/groupA/service/BandCompServiceTest.java:55]: this test doesn't test any of our code logic that isn't tested by the test above so can be removed

File: src/test/java/org/kainos/groupA/integration/BandCompIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    @Test
+    void getCompByBandID_shouldReturnStatusCode404() {
+        Response response = APP.client().target("http://localhost:8080/api/band-comp/string")
+                .request()
+                .get(Response.class);
+        Assertions.assertEquals(response.getStatus(), HttpStatus.NOT_FOUND_404);
+    }
+
+    @Test
+    void getCompByBandID_shouldReturnListOfCompetenciesForBand() {
```
Comment (shaunganley) [inline src/test/java/org/kainos/groupA/integration/BandCompIntegrationTest.java:43]: you should merge this test with getCompByBandID_shouldReturnStatusCode200 and check the code and the response values as part of 1 test to save time

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/main/java/org/kainos/groupA/models/Competency.java (hunk 2 above)
Comment (PatrykKiszelewski) [reply src/main/java/org/kainos/groupA/models/Competency.java:28]: band name added to competency model as well as is returned from query

File: src/test/java/org/kainos/groupA/integration/BandCompIntegrationTest.java (hunk 4 above)
Comment (PatrykKiszelewski) [reply src/test/java/org/kainos/groupA/integration/BandCompIntegrationTest.java]: test merge with getCompByBandID_shouldReturnStatusCode200

**Review by shaunganley — APPROVED** [review]

**Pull request merged**